sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/state/MeterSharedState.java

```diff
@@ -98,39 +99,49 @@ public final WriteableMetricStorage registerSynchronousMetricStorage(
   }
 
   /** Registers new asynchronous storage associated with a given {@code long} instrument. */
-  public final void registerLongAsynchronousInstrument(
-      InstrumentDescriptor instrument,
-      MeterProviderSharedState meterProviderSharedState,
-      Consumer<ObservableLongMeasurement> metricUpdater) {
-
-    meterProviderSharedState
-        .getViewRegistry()
-        .findViews(instrument, getInstrumentationLibraryInfo())
-        .stream()
-        .map(
-            view ->
-                AsynchronousMetricStorage.longAsynchronousAccumulator(
-                    view, instrument, metricUpdater))
-        .filter(m -> !m.isEmpty())
-        .forEach(this::register);
+  public final List<AsynchronousMetricStorage<?, ObservableLongMeasurement>>
+      registerLongAsynchronousInstrument(
+          InstrumentDescriptor instrument,
+          MeterProviderSharedState meterProviderSharedState,
+          Consumer<ObservableLongMeasurement> callback) {
+
+    List<AsynchronousMetricStorage<?, ObservableLongMeasurement>> storages =
+        meterProviderSharedState
+            .getViewRegistry()
+            .findViews(instrument, getInstrumentationLibraryInfo())
+            .stream()
+            .map(view -> AsynchronousMetricStorage.createLongAsyncStorage(view, instrument))
+            .filter(m -> !m.isEmpty())
+            .map(this::register)
```

Comment:
Not a huge deal but `map` having side effects can lead to some potential confusion

Reply:
I agree. I think the alternative is worth it despite the additional code.